### ISSUE TITLE
Unit test long time blocked

### ISSUE THREAD

**Issue by shibd** [apache/pulsar-client-cpp#238]
https://github.com/apache/pulsar-client-cpp/actions/runs/4553149127/jobs/8047001376?pr=207

<img width="1726" alt="image" src="https://user-images.githubusercontent.com/33416836/228761420-368fc623-821b-4d5d-bc04-bcee01f92f49.png">

I try revert #232, is can be passed.

**Comment by shibd** [comment]
It seems not related with #232, This is flaky test: ConsumerInterceptorsTest.TestNegativeAcksSend

When test blocked, stack is:

```
    7486 Thread_3355006   DispatchQueue_1: com.apple.main-thread  (serial)
    + 7486 start  (in dyld) + 2544  [0x18a8ffe50]
    +   7486 main  (in pulsar-tests) + 44  [0x100be82b0]  main.cc:23
    +     7486 RUN_ALL_TESTS()  (in pulsar-tests) + 16  [0x100be82cc]  gtest.h:2293
    +       7486 testing::UnitTest::Run()  (in pulsar-tests) + 216  [0x10121b194]
    +         7486 testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*)  (in pulsar-tests) + 96  [0x10121b2a4]
    +           7486 testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl*, bool (testing::internal::UnitTestImpl::*)(), char const*)  (in pulsar-tests) + 124  [0x101246b78]
    +             7486 testing::internal::UnitTestImpl::RunAllTests()  (in pulsar-tests) + 1032  [0x10121b8d4]
    +               7486 testing::TestSuite::Run()  (in pulsar-tests) + 324  [0x10120e078]
    +                 7486 testing::TestInfo::Run()  (in pulsar-tests) + 304  [0x10120cf48]
    +                   7486 testing::Test::Run()  (in pulsar-tests) + 192  [0x10120c038]
    +                     7486 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*)  (in pulsar-tests) + 96  [0x10120c0e8]
    +                       7486 testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*)  (in pulsar-tests) + 124  [0x101241cec]
    +                         7486 ConsumerInterceptorsTest_TestNegativeAcksSend_Test::TestBody()  (in pulsar-tests) + 1284  [0x10095e518]  InterceptorsTest.cc:431
    +                           7486 pulsar::Producer::send(pulsar::Message const&)  (in pulsar-tests) + 260  [0x100fdb7f0]  Producer.cc:44
    +                             7486 pulsar::Future<pulsar::Result, pulsar::MessageId>::get(pulsar::MessageId&)  (in pulsar-tests) + 96  [0x10083b040]  Future.h:69
    +                               7486 std::condition_variable::wait(std::unique_lock<std::mutex>&)  (in libc++.1.dylib) + 28  [0x18ab76ac4]
    +                                 7486 _pthread_cond_wait  (in libsystem_pthread.dylib) + 1232  [0x18ac2a638]
    +                                   7486 __psynch_cvwait  (in libsystem_kernel.dylib) + 8  [0x18abed84c]
```
[stack.txt](https://github.com/apache/pulsar-client-cpp/files/11109257/stack.txt)

**Comment by shibd** [comment]
fix by #239
